api/src/main/java/io/minio/CopyConditions.java

```diff
@@ -0,0 +1,104 @@
+/* Minio Java Library for Amazon S3 Compatible Cloud Storage,
+ * (C) 2016, 2017 Minio,Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Collections;
+import java.util.Locale;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import io.minio.errors.InvalidArgumentException;
+
+/**
+ * A container class to hold all the Conditions to be checked
+ * before copying an object.
+ */
+public class CopyConditions {
+
+  private Map<String, String> copyConditions = new HashMap<>();
+
+  private static final DateTimeFormatter RFC1123_DATE_TIME_FORMATTER =
+      DateTimeFormat.forPattern("EEE, dd MMM yyyy HH:mm:ss 'GMT'")
+      .withZoneUTC().withLocale(Locale.US);
+
+  /**
+   * Set modified condition, copy object modified since given time.
+   *
+   * @throws InvalidArgumentException
+   *           When invalid argument is passed
+   */
+  public void setModified(DateTime date) throws InvalidArgumentException {
+    if (date == null) {
+      throw new InvalidArgumentException("Date cannot be empty");
+    }
+    copyConditions.put("x-amz-copy-source-if-modified-since", RFC1123_DATE_TIME_FORMATTER.print(date));
```

Comment:
We don't need to mention it as AMZ_DATE_FORMAT since it is not AWS standard - http.TimeFormat is a standard by itself so kept it as RFC1123.